docs/advanced-topics/advanced-losses-optimizers-and-poolers.md

```diff
@@ -0,0 +1,102 @@
+(advanced-losses-optimizers-poolers)=
+# {octicon}`mortar-board` Advanced Losses, Optimizers and Poolers
+Many of the models supported by Finetuner make use of similar methods throughout finetuning, such as the methods used for *calculating loss*, *sampling* and *pooling* . Finetuner offers alternatives to each of these methods, which can improve the performance of your finetuning run in some cases.
+
+## Loss Functions
+
+Loss functions are used to calculate the quality of embeddings while training; the higher the output of the loss function, the more the parameters of the model will be updated.
+By default we use `TripletMarginLoss`, however we support many other loss functions as well, including `ArcFaceLoss` and `CosFaceLoss`.
+
+```{Important}
+Please check the [developer reference](../../api/finetuner/#finetuner.fit) to get the available options for `loss`.
+```
+
+### TripletMarginLoss
+
+Let's first take a look at our default loss function, `TripletMarginLoss`.  
+
+`TripletMarginLoss` is a *contrastive* loss function, meaning that the loss is calculated by comparing the embeddings of multiple documents (3 to be exact) documents to each other.
+Each triplet of documents consists of an anchor document, a positive document and a negative document.
+The anchor and the positive document belong to the same class, and the negative document belongs to a different class.
+The goal of `TripletMarginLoss` is to maximise the difference between the distance from the anchor to the positive document, and the distance from the anchor to the negative document.
+For a more detailed explanation on Triplet Loss, as well as how samples are gathered to create these triplets, see {doc}`/advanced_topics/negative-mining`.
+
+### SphereFace losses
+
+SphereFace loss is a loss function that was first formulated for computer vision and face recognition tasks.
+Finetuner supports two variations of this loss function, `ArcFaceLoss` and `CosFaceLoss`.
+Instead of attempting to minimise the distance between positive pairs and maximise the distance between negative pairs, the SphereFace loss functions compare each sample with the center of each class
```

Comment:
```suggestion
Instead of attempting to minimise the distance between positive pairs and maximise the distance between negative pairs, the SphereFace loss functions compare each sample with the center of each class
```

I think here it is confusing what is the center of a class ?

In practice this is not really center but more estimated prototypes which is learn alongside the representation in the last linear softmax layer. Not sure how to put it in nice word so I let you the burden

Reply:
I have changed it to '...with an estimate of the center point of each classes' embeddings.'
I don't want to go into too much detail explaining it but i think this is more accurate